cardano-db-sync/src/Cardano/DbSync/Era/Shelley/Genesis.hs

```diff
@@ -268,15 +267,16 @@ insertTxOuts syncEnv trce blkId (TxIn txInId _, txOut) = do
             }
     DB.TxOutVariantAddress -> do
       addrDetailId <- insertAddressUsingCache cache UpdateCache addrRaw vAddress
-      void . DB.insertTxOut $ DB.VTxOutW (makeVTxOut addrDetailId txId) Nothing
+      void . DB.insertTxOut $ DB.VTxOutW (makeVTxOut addrDetailId) Nothing
   where
+    txId = DB.TxKey $ fromIntegral txIndex + 20000 * DB.unBlockKey blkId -- On mainnet genesis produces ~14000 txs
```

Comment:
maybe a little comment as to why we're not using the returned txId from running insertTx. I can't fully remember why :)

Reply:
Personal note to add
```
We're using app generated unique ids to avoid dependencies between tables, which can damage 
parallelism.
```